src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -111,10 +115,15 @@ Copyright (c) .NET Foundation. All rights reserved.
                             '$(EnableCompressionInSingleFile)' == 'true' And
                             '$(SelfContained)' != 'true'"
                  ResourceName="CompressionInSingleFileRequiresSelfContained" />
+    <NETSdkError Condition="'$(PublishAot)' == 'true' And
+                            '$(_SkipAotSupportedRidCheck)' != 'true' And
+                            $(_AotSupportedRids.Contains('$(RuntimeIdentifier)')) != 'true'"
```

Comment:
This will allow identifiers like `linux-arm` and arbitrary substrings. You might want to do `Contains(';$(RuntimeIdentifier);')` instead.